make check: compilation warning.

**Describe the bug**
After compiling, a "make check" spits out a warning.


[x] Did you check if this is a duplicate issue?
[x] Did you test it on the latest FRRouting/frr master branch?


**To Reproduce**
Steps to reproduce the behavior:
```
git clone https://github.com/frrouting/frr.git frr
cd frr
git reset --hard fd2edd5ac18eafeaecec147200337df1414ecda8
./bootstrap.sh
./configure \
    --prefix=/usr \
    --includedir=\${prefix}/include \
    --enable-exampledir=\${prefix}/share/doc/frr/examples \
    --bindir=\${prefix}/bin \
    --sbindir=\${prefix}/lib/frr \
    --libdir=\${prefix}/lib/frr \
    --libexecdir=\${prefix}/lib/frr \
    --localstatedir=/var/run/frr \
    --sysconfdir=/etc/frr \
    --with-moduledir=\${prefix}/lib/frr/modules \
    --with-libyang-pluginsdir=\${prefix}/lib/frr/libyang_plugins \
    --enable-multipath=64 \
    --enable-user=frr \
    --enable-group=frr \
    --enable-vty-group=frrvty \
    --enable-configfile-mask=0640 \
    --enable-logfile-mask=0640 \
    --enable-fpm \
    --enable-snmp=agentx \
    --enable-zeromq \
    --enable-protobuf \
    --enable-systemd \
    --with-pkg-git-qversion \
    --with-pkg-extra-version=-fd2edd5
make
make check
```

**Expected behavior**
No compilation warnings.

**Screenshots**
```
  CC       tests/lib/test_nexthop_iter-test_nexthop_iter.o
In function ‘str_append’,
    inlined from ‘str_appendf’ at tests/lib/test_nexthop_iter.c:55:2:
tests/lib/test_nexthop_iter.c:37:3: warning: ‘strncpy’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
   strncpy((*buf) + strlen(*buf), repr, strlen(repr) + 1);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests/lib/test_nexthop_iter.c: In function ‘str_appendf’:
tests/lib/test_nexthop_iter.c:37:40: note: length computed here
   strncpy((*buf) + strlen(*buf), repr, strlen(repr) + 1);
                                        ^~~~~~~~~~~~
```

**Versions**
 - OS Kernel: Debian 10.4 buster / Linux 4.19.0 amd64
 - FRR Version 7.5-dev

--- a/tests/lib/test_nexthop_iter.c
+++ b/tests/lib/test_nexthop_iter.c
@@ -19,9 +19,10 @@ static int verbose;
 static void str_append(char **buf, const char *repr)
 {
 	if (*buf) {
-		*buf = realloc(*buf, strlen(*buf) + strlen(repr) + 1);
+		size_t new_size = strlen(*buf) + strlen(repr) + 1;
+		*buf = realloc(*buf, new_size);
 		assert(*buf);
-		strncpy((*buf) + strlen(*buf), repr, strlen(repr) + 1);
+		(void)strlcat(*buf, repr, new_size);
 	} else {
 		*buf = strdup(repr);
 		assert(*buf);

